cpp/include/IceUtil/Timer.h

```diff
@@ -43,9 +44,10 @@ namespace IceUtil
         // is the timer thread, join the timer execution thread otherwise.
         void destroy();
 
-        // Schedule a task for execution after a given delay.
+        // Schedule or reschedule a task for execution after a given delay.
         template<class Rep, class Period>
-        void schedule(const TimerTaskPtr& task, const std::chrono::duration<Rep, Period>& delay)
+        void
+        schedule(const TimerTaskPtr& task, const std::chrono::duration<Rep, Period>& delay, bool cancelPrevious = false)
```

Comment:
Replaced cancelPrevious by new reschedule function.

> We could also get rid of the inlining.

All these functions are inline because they are template functions and therefore must be in the header. You prefer out-of-line template functions in the header?

Reply:
I missed the fact that it was a template, it's fine to keep them in the header.